alibi_detect/utils/fetching.py

```diff
@@ -470,7 +470,7 @@ def fetch_detector(filepath: Union[str, os.PathLike],
     Initialised pre-trained detector.
     """
     # create path (if needed)
-    filepath = Path(filepath).joinpath(detector_name)
+    filepath = Path(filepath)
```

Comment:
@jklaise perhaps this is the most straightforward fix? There is no confusing logic for the user to contend with. 

We would have to modify all the notebooks though e.g. delete the `filepath = os.path.join(filepath, detector_name)` line in `cd_distillation_cifar10.ipynb`.

Reply:
This is probably what the user would expect to happen so I agree.

Wrt notebooks, should we modify the saved paths to be `./models/detector_name` so the current behaviour is unchanged (i.e. running multiple notebooks saves all detectors in their own subdirectory so they can be retrieved later).

Reply:
That's a much nicer solution with the notebooks, I'll do that 👍🏻

Reply:
I've had a go at moving the `os.path.join(filepath,detector_name)` outside the `if load_pretrained:` block, so `filepath` should be consistent for fetch and save routes.